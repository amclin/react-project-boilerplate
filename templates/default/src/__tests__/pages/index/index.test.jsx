import React from 'react'
import { render } from '@testing-library/react'
import { useQuery } from '@apollo/react-hooks'
import App from '../../../pages/index'

jest.mock('@apollo/react-hooks')
useQuery.mockReturnValue({ loading: true })

describe('Pages:Homepage', () => {
  it('renders the homepage', () => {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })
})
