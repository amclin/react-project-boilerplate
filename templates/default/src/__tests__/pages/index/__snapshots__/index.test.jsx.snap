// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages:Homepage renders the homepage 1`] = `
<DocumentFragment>
  <div
    class="jsx-2781750882 text-center"
  >
    <p
      class="jsx-2781750882"
    >
      Welcome to the boilerplate React app using NextJS!
    </p>
    <p
      class="jsx-2781750882"
    >
      Try navigating to 
      <a
        href="/example/second-page"
      >
        another page
      </a>
       and observe the URL changing
    </p>
    <div
      class="country-selector"
      data-testid="loading-div"
    >
      Loading Countries...
    </div>
  </div>
</DocumentFragment>
`;
